lib/model/rwfolder.go

```diff
@@ -518,9 +520,11 @@ nextFile:
 		}
 
 		// Verify that the thing we are handling lives inside a directory,
-		// and not a symlink or empty space.
-		if !osutil.IsDir(f.dir, filepath.Dir(fi.Name)) {
-			f.newError(fi.Name, errNotDir)
+		// and not a symlink or empty space. If it's already deleted, handle
+		// the check inside deleteFile deleteDir only in the scenario where
+		// the subject we are supposed to delete actually exists.
+		if !fi.IsDeleted() && !osutil.IsDir(f.dir, filepath.Dir(fi.Name)) {
```

Comment:
This change is unnecessary. ([Line 515](https://github.com/AudriusButkevicius/syncthing/blob/f3dedf6a59822dbad2d12a4c85a0a58e32ec8c7f/lib/model/rwfolder.go#L515))